File: FrontEnd/lib/domain/interfaces/usecases/plants/i_get_plant_by_id_use_case.dart
/// @file i_get_plant_by_id_use_case.dart
/// @description Interfaz del caso de uso para obtener una planta por ID.
/// @module Plants
/// @layer Domain
library;

import '../../../entities/plant.dart';

abstract interface class IGetPlantByIdUseCase {
  /// Devuelve la planta con [plantId] o lanza AppError.notFound.
  Future<Plant> execute(String plantId);
}

File: BackEnd/src/domain/interfaces/usecases/plants/IGetPlantByIdUseCase.ts
/**
 * @file IGetPlantByIdUseCase.ts
 * @description Interfaz del caso de uso para obtener una planta por ID.
 * @module Plants
 * @layer Domain
 */

import type { PlantResponseDTO } from '../../../dtos/plants/plant-response.dto.js';

export interface IGetPlantByIdUseCase {
  /**
   * @param plantId — Id de la planta.
   * @param userId — Id del usuario (para verificar ownership).
   * @returns Planta como DTO de respuesta.
   * @throws {NotFoundException} Si la planta no existe.
   * @throws {UnauthorizedException} Si el usuario no es propietario.
   */
  execute(plantId: string, userId: string): Promise<PlantResponseDTO>;
}
